router/file.go

```diff
@@ -1,69 +1,158 @@
-	name := form.Value["name"][0]
-	requestID, err := uuid.Parse(form.Value["request_id"][0])
-	if err != nil {
-		return echo.NewHTTPError(http.StatusBadRequest, err)
-	}
-
-	mimetype := file.Header.Get(echo.HeaderContentType)
-	if !acceptedMimeTypes[mimetype] {
-		return c.NoContent(http.StatusUnsupportedMediaType)
-	}
-
-	src, err := file.Open()
-	if err != nil {
-		return echo.NewHTTPError(http.StatusInternalServerError, err)
-	}
-	defer src.Close()
-
-	err = h.Storage.Save(name, src)
-	if err != nil {
-		return echo.NewHTTPError(http.StatusInternalServerError, err)
-	}
-
-	// TODO: src とかを引数に取らなくて良いようにする
-	created, err := h.Repository.CreateFile(c.Request().Context(), src, name, mimetype, requestID)
-	if err != nil {
-		return err
-	}
-
-	fileIDs = append(fileIDs, &created.ID)
-	return c.JSON(http.StatusOK, &FileResponse{fileIDs})
-}
-
-func (h *Handlers) GetFile(c echo.Context) error {
-	return c.NoContent(http.StatusOK)
-	// TODO: Implement
-}
-
-func (h *Handlers) DeleteFile(c echo.Context) error {
-	return c.NoContent(http.StatusOK)
-	// TODO: Implement
-}
+package router
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/labstack/echo/v4"
+)
+
+type FileResponse struct {
+	ID uuid.UUID `json:"id"`
+}
+
+type FileMetaResponse struct {
+	ID        uuid.UUID `json:"id"`
+	Name      string    `json:"name"`
+	MimeType  string    `json:"mime_type"`
+	CreatedAt time.Time `json:"created_at"`
+}
+
+var acceptedMimeTypes = map[string]bool{
+	"image/jpeg":         true,
+	"image/png":          true,
+	"image/gif":          true,
+	"image/bmp":          true,
+	"application/pdf":    true,
+	"application/msword": true,
+	"application/zip":    true,
+}
+
+func (h *Handlers) PostFile(c echo.Context) error {
+	form, err := c.MultipartForm()
+	if err != nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, err)
+	}
+	files, ok := form.File["file"]
+	if !ok || len(files) != 1 {
+		return echo.NewHTTPError(http.StatusBadRequest, fmt.Errorf("invalid file"))
+	}
+	reqfile := files[0]
+	names, ok := form.Value["name"]
+	if !ok || len(names) != 1 {
+		return echo.NewHTTPError(http.StatusBadRequest, fmt.Errorf("invalid file name"))
+	}
+	name := names[0]
+	requestIDs, ok := form.Value["request_id"]
+	if !ok || len(requestIDs) != 1 {
+		return echo.NewHTTPError(http.StatusBadRequest, fmt.Errorf("invalid file request id"))
+	}
+	requestID, err := uuid.Parse(requestIDs[0])
+	if err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, err)
+	}
+
+	mimetype := reqfile.Header.Get(echo.HeaderContentType)
+	if !acceptedMimeTypes[mimetype] {
+		return echo.NewHTTPError(http.StatusUnsupportedMediaType, fmt.Errorf("unsupported media type"))
+	}
+
+	src, err := reqfile.Open()
+	if err != nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, err)
+	}
+	defer src.Close()
+
+	ctx := context.Background()
+	file, err := h.Repository.CreateFile(ctx, src, name, mimetype, requestID)
```

Comment:
ここにコンテキストがあるから使うと良さそう
```suggestion
	file, err := h.Repository.CreateFile(c.Request().Context(), src, name, mimetype, requestID)
```